internal/storage/db/db.go

```diff
@@ -0,0 +1,133 @@
+package db
+
+import (
+	"context"
+	"time"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+
+	"github.com/ramil063/firstgodiplom/internal/logger"
+)
+
+type Storage struct{}
+
+// DataBaser основные команды для работы с бд
+type DataBaser interface {
+	ExecContext(ctx context.Context, query string, args ...any) (pgconn.CommandTag, error)
+	QueryRowContext(ctx context.Context, query string, args ...any) pgx.Row
+	QueryContext(ctx context.Context, query string, args ...any) (pgx.Rows, error)
+	Open() (*pgxpool.Pool, error)
+	PingContext(ctx context.Context) error
+	SetPool() error
+}
+
+// Init инициализация таблиц бд и проверка соединения
+func Init(dbr DataBaser) error {
+	var err error
+
+	if err = CheckPing(dbr); err != nil {
+		logger.WriteErrorLog(err.Error())
+		return err
+	}
+
+	err = CreateTables(dbr)
+	return err
+}
+
+// CheckPing проверка соединения с бд
+func CheckPing(dbr DataBaser) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
+	return dbr.PingContext(ctx)
+}
+
+// CreateTables создание основных таблиц
+func CreateTables(dbr DataBaser) error {
+	var err error
+
+	createTablesSQL := `--USERS
+	CREATE TABLE IF NOT EXISTS public.users
+	(
+		id   		serial not null constraint pk_users_id primary key,
+		name  		varchar(254),
+		login    	varchar(100) not null constraint uidx_users_login unique,
+		password 	varchar(100) not null,
+		access_token            varchar(100),
+		access_token_expired_at integer
+	);
+	COMMENT ON COLUMN public.users.id IS 'Идентификатор пользователя';
+	COMMENT ON COLUMN public.users.name IS 'Имя пользователя';
+	COMMENT ON COLUMN public.users.login IS 'Логин пользователя';
+	COMMENT ON COLUMN public.users.password IS 'Пароль пользователя';
+	COMMENT ON COLUMN public.users.access_token IS 'Токен пользователя';
+	COMMENT ON COLUMN public.users.access_token_expired_at IS 'Токен пользователя истекает';
+
+	        --ORDER
+	CREATE TABLE IF NOT EXISTS public.order
+	(
+		id         	serial not null constraint pk_order_id primary key,
+		number     	varchar(64) not null constraint uidx_order_number unique,
+		accrual     double precision default 0 not null,
```

Comment:
```suggestion
		accrual     decimal default 0 not null,
```
или есть еще тип money

Reply:
поправил